dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Machines/VoltageRegulators/Standard/BaseClasses/RectifierRegulationCharacteristic.mo

```diff
@@ -16,30 +16,18 @@ within Dynawo.Electrical.Controls.Machines.VoltageRegulators.Standard.BaseClasse
 block RectifierRegulationCharacteristic "Characteristic function of rectifier regulation"
   extends Modelica.Blocks.Interfaces.SISO;
 
-  parameter Real UHigh = 0.75 "Upper limit of non-linear mode";
-  parameter Real ULow = sqrt(3) / 4 "Lower limit of non-linear mode";
+  final parameter Real[4] U = {0, sqrt(3) / 4, 0.75, 1} "Abscissas of the points defining the linear modes";
+  final parameter Real[4] Y = {1, 0.75, sqrt(3) / 4, 0} "Ordinates of the points defining the linear modes";
 
-  final parameter Real A1 = if ULow == 0 then 0 else (1 - sqrt(UHigh - ULow ^ 2)) / ULow "Characteristic coefficient of first linear mode";
-  final parameter Real A2 = if UHigh == 1 then 0 else sqrt(UHigh / (1 - UHigh)) "Characteristic coefficient of second linear mode";
+  Modelica.Blocks.Tables.CombiTable1Ds fRect(extrapolation = Modelica.Blocks.Types.Extrapolation.HoldLastPoint, table = [U, Y]);
 
 equation
-  assert(ULow <= UHigh, "Lower limit is lower than or equal to upper limit");
-
-  if u <= 0 then
-    y = 1;
-  elseif u > 0 and u <= ULow then
-    y = 1 - A1 * u;
-  elseif u > ULow and u < UHigh then
-    y = sqrt(UHigh - u ^ 2);
-  elseif u >= UHigh and u <= 1 then
-    y = A2 * (1 - u);
-  else
-    y = 0;
-  end if;
+  u = fRect.u;
+  y = noEvent(if (u > sqrt(3) / 4 and u < 0.75) then sqrt(0.75 - u ^ 2) else fRect.y[1]);
```

Comment:
I don't understand the advantage of replacing the actual model with this function, the curves.csv seem to have almost the same size

Reply:
For PmConstAc7bPss3b and PmConstAc7cPss2c, I get a reduction in the .csv size of more than 8 percent. For PmConstSt9cPss2c, this figure is 5 percent. For all other cases, the reduction is insignificant.

You can close the PR if you deem these gains to be too small.

Reply:
In these test cases, the gains are too small so I don't think it is necessary to make the changes now. Also, the size of the curves, as it is now, is acceptable for 15s.
I will close the PR and the issue for now. If we see other problems later, we can retry this solution.